neo/api/JSONRPC/JsonRpcApi.py

```diff
@@ -128,6 +129,21 @@ def parse_uint_str(self, param):
             return param[2:]
         return param
 
+    def validateaddress(self, params):
+        # check for [] parameter or [""]
+        if params is None or params[0] == '':
```

Comment:
this code doesn't yet check for an empty list (it is not None). perhaps upgrade to `if params is None or len(params) == 0 or params[0] == '':`

Reply:
I should have anticipated this as I initially was surprised as well.  However, this code is correct given the current `gen_rpc_req` implementation. Here's why

```
req = self._gen_rpc_req("validateaddress", params=[])
```
returns a `dict` without the `params` keyword, therefor `if params is None` actually tests for the empty list. I'm assuming an actual request will be parsed similarly.

```
In [3]: def test_list(params=None):
   ...:     if params:
   ...:         print("yes params")
   ...:     else:
   ...:         print("no params")
   ...:

In [4]: test_list([])
no params
```

Reply:
I understand what you are saying. but this is not 100% correct. `if not params` matches an empty list, `if params is None` doesn't. you can just replace this if here with `if not params or params[0] == '':` for the intended effect.